common/common.cpp

```diff
@@ -726,6 +726,15 @@ std::tuple<struct llama_model *, struct llama_context *> llama_init_from_gpt_par
         return std::make_tuple(nullptr, nullptr);
     }
 
+    if (params.n_threads == -2) {
+        params.n_threads = get_num_physical_cores();
+#ifdef GGML_USE_CUBLAS
+        if (params.n_gpu_layers >= llama_model_n_layer(model) + 3) {
+            params.n_threads = 1;
+        }
+#endif // GGML_USE_CUBLAS
```

Comment:
This is an implementation detail that the user should not need to know and in the future we will fix this anyway